R/estimateParams.R

```diff
@@ -53,3 +53,97 @@ prepareData <- function(data) {
 
   as.data.frame(dataNew[, -which(names(dataNew) == "time")], row.names = seq_len(nrow(dataNew)))
 }
+
+
+# transition <- weibull_transition(h01 = 1.5, h02 = 0.5, h12 = 1.1, p01 = 0.4, p02 = 0.5, p12 = 2.4)
+# trial <- getOneClinicalTrial(nPat = c(10000),
+#                             transitionByArm = list(transition),
+#                             dropout = list(rate = 0.3, time = 1),
+#                             accrual = list(param = "intensity", value = 100))
+
+# estimateParams(trial, "Weibull")
+
+NegLogLik <- function(transition, data) {
+  UseMethod(NegLogLik)
+}
+
+NegLogLik.ExponentialTransition <- function(transition, data) {
+  with(data, -sum(log(haz(transition, exit, trans)^status * SurvTrans(transition, exit, trans) / SurvTrans(transition, entry, trans))))
+}
+
+NegLogLik.WeibullTransition <- function(transition, data) {
+  with(data, -sum(log(haz(transition, exit, trans)^status * SurvTrans(transition, exit, trans) / SurvTrans(transition, entry, trans))))
+}
+
+haz <- function(transition, t, trans = c(1, 2, 3)) {
```

Comment:
here we really need the generic function. btw. How about the PWC method? don't we need that too?

Reply:
I have left out PWC for now since we said it should not be part of the scope of the parameter estimation update.